engine/src/main/java/org/hibernate/validator/internal/metadata/core/MetaConstraintBuilder.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.internal.metadata.core;
+
+import java.lang.annotation.Annotation;
+
+import org.hibernate.validator.internal.engine.valueextraction.ValueExtractorManager;
+import org.hibernate.validator.internal.metadata.descriptor.ConstraintDescriptorImpl;
+import org.hibernate.validator.internal.metadata.location.ConstraintLocation;
+import org.hibernate.validator.internal.properties.Constrainable;
+import org.hibernate.validator.internal.util.TypeResolutionHelper;
+
+/**
+ * @author Marko Bekhta
+ */
+public class MetaConstraintBuilder<A extends Annotation> {
+
+	private final ConstraintDescriptorImpl<A> constraintDescriptor;
+	private final ConstraintLocation.Builder locationBuilder;
```

Comment:
The part about location builder in this commit can be ignored. It will be removed in later commits. And I think about removing this builder as well. I think that storing the ConstraintDefs will be enough and this logic will be moved elsewhere.